lib/Target/DebugInfo/EmitHGLDD.cpp

```diff
@@ -106,6 +107,51 @@ struct DenseMapInfo<JValue> {
 };
 } // namespace llvm
 
+/// Make the given `path` relative to the `relativeTo` path and store the result
+/// in `relativePath`. Returns whether the conversion was successful. Fails if
+/// the `relativeTo` path has a longer prefix of `../` than `path`, or if it
+/// contains any non-prefix `../` components. Does not clear `relativePath`
+/// before appending to it.
+static bool makePathRelative(StringRef path, StringRef relativeTo,
```

Comment:
I wonder what the llvm policy is on using std::filesystem::path (https://en.cppreference.com/w/cpp/filesystem/path)

Reply:
Which handles a few more kinds of "interesting" paths, like "/////".

Reply:
Awesome! I hadn't realized C++ finally gained a standard way of dealing with file paths :partying_face: